frame/contracts/src/weights.rs

```diff
@@ -150,1183 +150,1191 @@ pub trait WeightInfo {
+		(109_742_000 as Weight)
+			// Standard Error: 101_000
+			.saturating_add((130_236_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 	}
 	fn seal_input(r: u32, ) -> Weight {
-		(118_826_000 as Weight)
-			// Standard Error: 89_000
-			.saturating_add((6_650_000 as Weight).saturating_mul(r as Weight))
+		(119_749_000 as Weight)
+			// Standard Error: 71_000
+			.saturating_add((7_019_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 	}
 	fn seal_input_per_kb(n: u32, ) -> Weight {
-		(132_497_000 as Weight)
+		(130_185_000 as Weight)
 			// Standard Error: 0
-			.saturating_add((278_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add((276_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 	}
 	fn seal_return(r: u32, ) -> Weight {
-		(112_447_000 as Weight)
-			// Standard Error: 73_000
-			.saturating_add((4_398_000 as Weight).saturating_mul(r as Weight))
+		(113_866_000 as Weight)
+			// Standard Error: 90_000
+			.saturating_add((4_778_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 	}
 	fn seal_return_per_kb(n: u32, ) -> Weight {
-		(120_288_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((787_000 as Weight).saturating_mul(n as Weight))
+		(123_666_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((780_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 	}
 	fn seal_terminate(r: u32, ) -> Weight {
-		(118_973_000 as Weight)
-			// Standard Error: 124_000
-			.saturating_add((75_967_000 as Weight).saturating_mul(r as Weight))
+		(120_302_000 as Weight)
+			// Standard Error: 166_000
+			.saturating_add((94_079_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().writes((4 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().writes((5 as Weight).saturating_mul(r as Weight)))
 	}
 	fn seal_restore_to(r: u32, ) -> Weight {
-		(207_295_000 as Weight)
-			// Standard Error: 385_000
-			.saturating_add((103_584_000 as Weight).saturating_mul(r as Weight))
+		(212_378_000 as Weight)
+			// Standard Error: 381_000
+			.saturating_add((126_862_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
-			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().writes((4 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().writes((6 as Weight).saturating_mul(r as Weight)))
```

Comment:
The refcount of the of the PrefabWasmModule needs to be incremented. Previously, the code of the target contract wasn't accessed at all.